aws-inventory/lambda/inventory-firehose.py

```diff
@@ -72,16 +72,32 @@ def discover_firehose(target_account, region):
 
     firehoses = []
     client = target_account.get_client('firehose', region=region)
-    response = client.list_delivery_streams()
-    while 'HasMoreDeliveryStreams' in response and response['HasMoreDeliveryStreams'] is True:  # Gotta Catch 'em all!
-        firehoses += response['DeliveryStreamNames']
-        response = client.list_delivery_streams(ExclusiveStartDeliveryStreamName=response['DeliveryStreamNames'][-1])
-    firehoses += response['DeliveryStreamNames']
-
-    logger.debug(f"Discovered {len(firehoses)} Firehose Delivery Streams in {target_account.account_name}")
+    try:
+        response = client.list_delivery_streams()
+    except ClientError as error:
+        logger.error(error)
+    while response['HasMoreDeliveryStreams'] == True:  # Gotta Catch 'em all!
```

Comment:
Are we sure HasMoreDeliveryStreams is always in response? My original code has a "'HasMoreDeliveryStreams' in response and" clause.

Reply:
HasMoreDeliveryStreams will always be in the request with a true or false value. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/firehose.html#Firehose.Client.list_delivery_streams